src/cloud-sql-instance.ts

```diff
@@ -106,17 +107,45 @@ export class CloudSQLInstance {
     }) as ReturnType<typeof pThrottle>;
   }
 
-  forceRefresh(): Promise<RefreshResult> {
+  forceRefresh(): Promise<void> {
     // if a refresh is already ongoing, just await for its promise to fulfill
     // so that a new instance info is available before reconnecting
     if (this.next) {
-      return this.next;
+      return new Promise(resolve => {
+        if (this.next) {
+          this.next.finally(resolve);
+        } else {
+          resolve();
+        }
+      });
     }
+
     this.cancelRefresh();
-    return this.refresh();
+    this.scheduleRefresh(0);
+
+    return new Promise(resolve => {
+      // setImmediate() to yield execution to allow other refresh background
```

Comment:
this comment references `setImmediate` but we are using `setTimeout`...

Reply:
Fixed. setTimeout() is the correct function.